src/main/scala/com/monsanto/arch/cloudformation/model/resource/Lambda.scala

```diff
@@ -92,3 +92,47 @@ object `AWS::Lambda::EventSourceMapping` extends DefaultJsonProtocol {
   implicit val format: JsonFormat[`AWS::Lambda::EventSourceMapping`] = jsonFormat7(`AWS::Lambda::EventSourceMapping`.apply)
 }
 
+case class `AWS::Lambda::Version`(name: String,
+                                  FunctionName: Token[ResourceRef[`AWS::Lambda::Function`]],
+                                  Description: Option[String] = None,
+                                  CodeSha256: Option[Token[String]] = None,
+                                  override val Condition: Option[ConditionRef] = None)
+    extends Resource[`AWS::Lambda::Version`]
+    with HasArn {
+
+  override def arn = `Fn::GetAtt`(Seq(name, "Arn"))
+
+  def version = `Fn::GetAtt`(Seq(name, "Version"))
+
+  def when(newCondition: Option[ConditionRef] = Condition) = copy(Condition = newCondition)
+}
+
+object `AWS::Lambda::Version` extends DefaultJsonProtocol {
+  implicit val format: JsonFormat[`AWS::Lambda::Version`] = jsonFormat5(
+      `AWS::Lambda::Version`.apply)
+}
+
+case class `AWS::Lambda::Alias`(name: String,
+                                AliasName: Token[String],
+                                FunctionName: Token[ResourceRef[`AWS::Lambda::Function`]],
+                                FunctionVersion: Token[String],
```

Comment:
I wonder if this should be a ResourceRef[`AWS::Lambda::Version`]. If we ever needed to pass it in as a parameter we could make a special type for that as well (see something like #84).